pkg/op/storage.go

```diff
@@ -27,6 +27,10 @@ type AuthStorage interface {
 	GetKeySet(context.Context) (*jose.JSONWebKeySet, error)
 }
 
+type ClientCredentialsXXX interface {
```

Comment:
I've added this new method that the storage provider can implement.

The _ClientCredentialsXXX_ interface is separate from the OPStorage and AuthStorage interfaces so that this PR doesn't introduce a breaking change.

The method is used to allow the storage provider (library consumer) to control the subject/aud/scopes on the access token created during the flow.

_more comments on this below_